types/node/index.d.ts

```diff
@@ -4007,7 +4007,10 @@ declare module "assert" {
             });
         }
 
-        export function fail(actual: any, expected: any, message: string, operator: string): void;
+        export var fail: {
```

Comment:
It's important to declare as overloads so other libraries can augment the definition.